pkg/app/piped/notifier/slack.go

```diff
@@ -151,6 +152,19 @@ func (s *slack) buildSlackMessage(event model.NotificationEvent, webURL string)
 			{"Started At", makeSlackDate(d.CreatedAt), true},
 		}
 	}
+	generateDeploymentEventDataForTriggerFailed := func(app *model.Application, hash, msg string) {
+		link = fmt.Sprintf("%s/applications/%s?project=%s", webURL, app.Id, app.PipedId)
+		commitURL, err := git.MakeCommitURL(app.GitPath.Repo.Remote, hash)
+		if err != nil {
+			s.logger.Error(fmt.Sprintf("failed to get the URL for the specified commit: %v", err))
+		}
+		fields = []slackField{
+			{"Project", truncateText(app.ProjectId, 8), true},
+			{"Application", makeSlackLink(app.Name, link), true},
+			{"Kind", strings.ToLower(app.Kind.String()), true},
+			{"Commit", makeSlackLink(truncateText(msg, 8), commitURL), true},
```

Comment:
If `commitURL` is empty, this entry should be hidden.

Reply:
5028483